src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -103,6 +114,20 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return (typename SGVector<T>::EigenVectorXtMap(a)).dot(typename SGVector<T>::EigenVectorXtMap(b));
 	}
 
+	/** Eigen3 vector sum method */
+	template <typename T>
+	T sum_impl(const SGVector<T>& vec) const
+	{
+		return (typename SGVector<T>::EigenVectorXtMap(vec)).sum();
+	}
+
+	/** Eigen3 matrix sum method */
+	template <typename T>
+	T sum_impl(const SGMatrix<T>& mat) const
```

Comment:
can we please have row-wise and column wise sum here as well?
Should that be in a new patch or not?